chainer/__init__.py

```diff
@@ -169,6 +169,7 @@ def is_arrays_compatible(arrays):
     raise TypeError('incorrect dtype name in CHAINER_DTYPE: "{}". '
                     'Only float16/32/64 are allowed.'.format(_chainer_dtype))
 global_config.dtype = numpy.dtype(_chainer_dtype)
+global_config.recompute = False
```

Comment:
* The configuration key `recompute` sounds too general and a bit confusing to users. How about naming it like `in_recomputation`?
* Could you add the description like this to https://docs.chainer.org/en/latest/reference/configuration.html#configuration-keys?

```rst
This flag is automatically set by :func:`chainer.functions.forget` and not intended to be changed by users.
You can use this flag when implementing your own Link to avoid updating the internal states during recomputation done by `F.forget`.
See the documentation of :func:`chainer.functions.forget` for details.
```

Reply:
Thank you for review. I fixed code.